src/Integration.Vsix/Suppression/SuppressionManager.cs

```diff
@@ -94,12 +94,20 @@ private bool ShouldIssueBeSuppressed(Diagnostic diagnostic)
 
             // TODO: ?need to make file path relative to the project file path
             // As a minimum, the project, file and rule id must match
-            var issuesInFile = sonarqubeIssueProvider.GetSuppressedIssues(liveIssue.ProjectGuid, liveIssue.IssueFilePath)
-                    .Where(i => StringComparer.OrdinalIgnoreCase.Equals(liveIssue.Diagnostic.Id, i.RuleId)); // TODO: rule repository?
+            var issuesInFile = sonarqubeIssueProvider.GetSuppressedIssues(liveIssue.ProjectGuid, liveIssue.IssueFilePath);
 
-            return issuesInFile.Any(i =>
+            if (issuesInFile == null)
+            {
+                return true;
+            }
+
+            // TODO: rule repository?
+            issuesInFile = issuesInFile.Where(i => StringComparer.OrdinalIgnoreCase.Equals(liveIssue.Diagnostic.Id, i.RuleId));
+            
+            bool matchFound = issuesInFile.Any(i =>
                     liveIssue.StartLine == i.Line ||
                     StringComparer.Ordinal.Equals(liveIssue.LineHash, i.Hash));
+            return matchFound;
```

Comment:
I split some of the statements up to make debugging easier (since the enums are lazily evaluated it was difficult to work out what was being returned at each stage).

Reply:
We might want to invest into [OzCode](https://oz-code.com/) at some point, it makes debugging Linq expression really easy.